### PR TITLE
Run Delta Lake tests with Spark 3.2.x

### PR DESCRIPTION
**Pull request by andygrove**
Part of https://github.com/NVIDIA/spark-rapids/issues/5966

The goal of this PR is to run the Delta Lake integration tests against Spark 3.2.1 using the open-source version of Delta Lake.

### CODE REVIEW

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
build

File: jenkins/spark-tests.sh [hunk 1]
```diff
@@ -161,10 +161,26 @@ export CUDF_UDF_TEST_ARGS="--conf spark.rapids.memory.gpu.allocFraction=0.1 \
...
 
+
 export SCRIPT_PATH="$(pwd -P)"
 export TARGET_DIR="$SCRIPT_PATH/target"
 mkdir -p $TARGET_DIR
 
+run_delta_lake_tests() {
+  echo "run_delta_lake_tests SPARK_VER = $SPARK_VER"
+  SPARK_321_PATTERN="(32[1-9])"
+  DELTA_LAKE_VER = "1.2.1"
```
Comment (jlowe) [inline jenkins/spark-tests.sh:172]:
```suggestion
  DELTA_LAKE_VER="1.2.1"
```

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
build

**Comment by jlowe** [comment]
I was, in turn, going off of what cudf-udf was doing.  TEST_MODE defaults to IT_ONLY, so not surprising it doesn't run during "normal" runs.  I assumed we were specifying "ALL" during our nightly tests, but it looks like we may not have a build that runs for ALL?  Wondering if we should update these to also run for TEST_MODE=IT_ONLY.  cc: @pxLi

File: jenkins/spark-tests.sh [hunk 2]
```diff
@@ -305,6 +325,11 @@ if [[ "$TEST_MODE" == "ALL" || "$TEST_MODE" == "CUDF_UDF_ONLY" ]]; then
   run_test_not_parallel cudf_udf_test
 fi
 
+# Delta Lake tests
+if [[ "$TEST_MODE" == "ALL" || "$TEST_MODE" == "DELTA_LAKE_ONLY" ]]; then
```
Comment (tgravescs) [inline jenkins/spark-tests.sh:329]: nit update comment on line 251 for test mode

File: jenkins/spark-tests.sh [hunk 3]
```diff
@@ -161,10 +161,26 @@ export CUDF_UDF_TEST_ARGS="--conf spark.rapids.memory.gpu.allocFraction=0.1 \
...
 --py-files ${RAPIDS_PLUGIN_JAR}"
 
+
 export SCRIPT_PATH="$(pwd -P)"
 export TARGET_DIR="$SCRIPT_PATH/target"
 mkdir -p $TARGET_DIR
 
+run_delta_lake_tests() {
+  echo "run_delta_lake_tests SPARK_VER = $SPARK_VER"
+  SPARK_321_PATTERN="(3\.2\.[1-9])"
```
Comment (tgravescs) [inline jenkins/spark-tests.sh:171]: minor side note, CDH is 3.2.1 based but I don't know if it has delta lake with it, assume separate issue anyway but just mentioning

**Comment by andygrove** [comment]
> I was, in turn, going off of what cudf-udf was doing. TEST_MODE defaults to IT_ONLY, so not surprising it doesn't run during "normal" runs. I assumed we were specifying "ALL" during our nightly tests, but it looks like we may not have a build that runs for ALL? Wondering if we should update these to also run for TEST_MODE=IT_ONLY. cc: @pxLi

Hi @pxLi could you help provide some direction with this?

**Comment by jlowe** [comment]
The nightly tests setup was changed in #6008.  It now uses `DEFAULT` instead of `IT_ONLY` and `ALL`.  So you should be able to check for `DEFAULT` or `DELTA_LAKE_ONLY`.

**Comment by pxLi** [comment]
> > I was, in turn, going off of what cudf-udf was doing. TEST_MODE defaults to IT_ONLY, so not surprising it doesn't run during "normal" runs. I assumed we were specifying "ALL" during our nightly tests, but it looks like we may not have a build that runs for ALL? Wondering if we should update these to also run for TEST_MODE=IT_ONLY. cc: @pxLi
> 
> Hi @pxLi could you help provide some direction with this?

Yes, Jason is right. Use `DEFAULT` (will be run in regular nightly IT pipelines) or `DELTA_LAKE_ONLY`

**Comment by andygrove** [comment]
> Yes, Jason is right. Use `DEFAULT` (will be run in regular nightly IT pipelines) or `DELTA_LAKE_ONLY`

Thanks @pxLi  I have made those changes. Do we have the ability to run nightly CI from this branch/PR so that we can verify that it works before merging?

**Comment by pxLi** [comment]
> > Yes, Jason is right. Use `DEFAULT` (will be run in regular nightly IT pipelines) or `DELTA_LAKE_ONLY`
> 
> Thanks @pxLi I have made those changes. Do we have the ability to run nightly CI from this branch/PR so that we can verify that it works before merging?

The script relies on pre-built artifacts, 
currently we may just comment out the mvn artifacts getter part and use local build jars

**Comment by andygrove** [comment]
Thanks @pxLi. I spent some time hacking the scripts to try and test with local jars but wasn't confident that I was doing this correctly. 

I did some more manual testing both locally and on Databricks and I think this is safe to merge.

**Comment by andygrove** [comment]
@jlowe @tgravescs Could I get a review when you have time. I think this is safe to merge now.

**Review by jlowe — APPROVED** [review]

**Comment by andygrove** [comment]
build

**Review by pxLi — APPROVED** [review]

**Pull request merged**